tests/resource/test_resource_setters.py

```diff
@@ -18,10 +18,258 @@ class TestResourceSetters(unittest.TestCase):
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_access_level(
+                "TESTING",
+                "ELIJTEST",
+                alter="success",
+                control="failure",
+                update="all",
+                read="all",
+            ),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ACCESS_LEVEL_ALL_REQUEST_XML,
+        )
+
+    @patch("pyracf.resource.resource_admin.ResourceAdmin.extract")
+    def test_resource_admin_build_alter_audit_by_attempt_request(
+        self,
+        resource_admin_extract_mock: Mock,
+    ):
+        resource_admin_extract_mock.return_value = (
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_attempt(
+                "TESTING", "ELIJTEST", failure="control"
+            ),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ATTEMPT_REQUEST_XML,
+        )
+
+    @patch("pyracf.resource.resource_admin.ResourceAdmin.extract")
+    def test_resource_admin_build_alter_audit_by_attempt_none_request(
+        self,
+        resource_admin_extract_mock: Mock,
+    ):
+        resource_admin_extract_mock.return_value = (
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_attempt("TESTING", "ELIJTEST"),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ATTEMPT_NONE_REQUEST_XML,
+        )
+
+    @patch("pyracf.resource.resource_admin.ResourceAdmin.extract")
+    def test_resource_admin_build_alter_audit_by_attempt_multiple_request(
+        self,
+        resource_admin_extract_mock: Mock,
+    ):
+        resource_admin_extract_mock.return_value = (
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_attempt(
+                "TESTING", "ELIJTEST", failure="control", all="read"
+            ),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ATTEMPT_MULT_REQUEST_XML,
+        )
+
+    @patch("pyracf.resource.resource_admin.ResourceAdmin.extract")
+    def test_resource_admin_build_alter_audit_by_attempt_all_request(
+        self,
+        resource_admin_extract_mock: Mock,
+    ):
+        resource_admin_extract_mock.return_value = (
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_GET_AUDIT_RULES
+        )
+        self.assertEqual(
+            self.resource_admin.alter_audit_by_attempt(
+                "TESTING", "ELIJTEST", failure="control", success="alter", all="read"
+            ),
+            TestResourceConstants.TEST_RESOURCE_ALTER_AUDIT_BY_ATTEMPT_ALL_REQUEST_XML,
+        )
+
+    def test_resource_admin_build_alter_audit_by_attempt_value_error(self):
```

Comment:
We should probably keep the `ValueError` tests together, no?

Reply:
I viewed this less as a testing of the value error functionality and more as a test of the `alter_audit_by_attempt` function to ensure it produces a value error under the right conditions. Functionally it tests both, so i think a good case can be made for either. (I did one for alter and one for overwrite).

Reply:
I think that is fine. Only other suggestion that I would have is to maybe create a test the second case on the switch statement in `__validate_access_levels`.

Reply:
makes sense. New test created for this purpose tied to `alter_audit_by _attempt`.